x/auth/ante/setup.go

```diff
@@ -40,6 +40,14 @@ func (sud SetUpContextDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 
 	newCtx = SetGasMeter(simulate, ctx, gasTx.GetGas())
 
+	if cp := ctx.ConsensusParams(); cp.Block != nil {
```

Comment:
i dont think this fixes the issue at hand, the mempool could still put in an amount of txs that is larger the maxgas.

Reply:
No, because txs are only inserted into the mempool if CheckTx passes. CheckTx would fail here so such a tx would never enter the app's mempool.

However, we also need to update the SDK's default ABCI++ proposal handlers so we can handle an entire proposal.